packages/rest/src/rest.server.ts

```diff
@@ -150,8 +150,14 @@ export class RestServer extends Context implements Server, HttpServerLike {
     return this._httpServer ? this._httpServer.listening : false;
   }
 
+  /**
+   * The base url for the server, including the basePath if set
+   */
   get url(): string | undefined {
-    return this._httpServer && this._httpServer.url;
+    let serverUrl = this._httpServer && this._httpServer.url;
+    if (!serverUrl) return serverUrl;
+    serverUrl = serverUrl + (this._basePath || '');
```

Comment:
I find it a little strange that a property named `url` in `RestServer` should have anything appended to it.

Shouldn't `basePath` be a setter-getter, instead?

Reply:
Are you proposing to have getters for base url and basePath? Url can have path and it matches servers in OpenAPI spec.

Reply:
See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#server-object

Reply:
Oh, I see.